java/src/jmri/managers/AbstractManager.java

```diff
@@ -179,6 +179,14 @@ public void deleteBean(@Nonnull E bean, @Nonnull String property) throws Propert
     public void register(E s) {
         String systemName = s.getSystemName();
 
+        E existingBean = getBeanBySystemName(systemName);
+        if (existingBean != null) {
```

Comment:
These two checks should be a single `if (existingBean != null && !e.equals(existingBean)) { ...`